internal/common/foraging/deerhunt_test.go

```diff
@@ -0,0 +1,61 @@
+package foraging
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+)
+
+var huntParticipants = map[shared.ClientID]float64{shared.Team1: 1.0, shared.Team2: 0.9} // arbitrarily chosen for test
+var params = DeerHuntParams{P: 0.95, Lam: 1.0}
+var hunt = DeerHunt{Participants: huntParticipants, Params: params}
+
+func TestDeerUtilityTier(t *testing.T) {
+
+	decay := 0.8 // use an arbitrary decay param
+	maxDeer := 4
+
+	var tests = []struct {
+		inputR float64 // cumulative resource input from hunt participants
+		want   int     // output tier
+	}{
+		{0.0, 0},
+		{0.99, 0},
+		{1.0, 1},
+		{1.8, 2},
+		{2.45, 3},
+		{2.96, 4},
+		{1000.0, 4},
+	}
+	for i, tt := range tests {
+		testname := fmt.Sprintf("%.3f", tt.inputR)
+		t.Run(testname, func(t *testing.T) {
+			ans := deerUtilityTier(tt.inputR, uint(maxDeer), decay)
+			fmt.Println(i)
+			if ans != uint(tt.want) {
+				t.Errorf("got %d, want %d", ans, tt.want)
+			}
+		})
+	}
+}
+
+func TestTotalInput(t *testing.T) {
+	ans := hunt.TotalInput()
+	if ans != 1.9 {
+		t.Errorf("TotalInput() = %.2f; want 1.9", ans)
+	}
+}
+
+func TestDeerReturn(t *testing.T) {
```

Comment:
this is extremely smart, but I'm not a huge fan of this as it might mean test signals are flaky.

The func seems simple enough to not need testing, so I think you can remove this.

Another way is via dependency injection, where you pass in `W` and `D` via arguments and mock them to return random values.

Again, that's not even a perfect test plan. Maybe there's is a way to seed those so that there is a deterministic value returned?

Non-issue here, delete the test if you see it fit :)